React/creating-components-2.md

```diff
@@ -0,0 +1,138 @@
+
+## Setup to manage list of data
+One student is great and all, but ideally, we'd really be able to manage our whole list of students that we might need.
+
+We'll create a new component called `StudentCollection` that will live between the existing  `App` and `Student` components. This new component will manage the full list of students and then render the data for each student.
+
+Spend a few minutes now going back to the [creating components](creating-components.md) notes to see how to create the new component file with the basic class, imports and exports you need.
+
+Move the code that renders `Student` from `App` to `StudentCollection`. Note: You will also need to move and update the `import` statement.
+
+Import and render `StudentCollection` now from the `App` component instead.
+
+[The code](https://github.com/AdaGold/react-hello-world/tree/part-4/src)
+
+
+## Use `state` for data
+
+Set up a constructor in the `StudentCollection` component which will use a variable to create a state-managed collection of student information. Don't forget your call to `super();` which is always required in a component constructor!
+
+```javascript
+constructor() {
+    super();
+
+    this.state = {
+      students: [
+        {
+          name: "State Ada",
+          email: "state-ada@ada.co"
+        },
+        {
+          name: "Cool",
+          email: "cool@ada.co"
+        }
+      ]
+    };
+  }
+```
+
+Next, we'll use a loop in our `render` function to iterate through each item in our state object and render a `Student` component for each piece of data. Remember that the `render` function needs to return a single element.
+
+```javascript
+render() {
+    let studentComponents = this.state.students.map(function(student) {
+      return <Student name={ student.name } email={ student.email }/>
+    });
+
+    return (
+      <div>
+        <h3>Students</h3>
+        { studentComponents }
+      </div>
+    )
+  }
+```
+
+Note that there are a few different ways to accomplish this same result. The approach we've taken here is to create a new variable _outside_ the `return` to construct the individual components. Then, within the return, we wrap the variable from above in a single `div` since we can only return one high-level element.
+
+Verify that you now see the data from the constructor in your browser. If you need to see what has gone wrong, take a look at [the code](https://github.com/AdaGold/react-hello-world/blob/part-5/src/components/student_collection.js).
+
+**Dive In:** Unpack the line with `return <Student...` with the person sitting next to you. Identify each piece of code on this line. Some questions to think about as you go through this activity:
+- What do you think the `studentComponents` variable looks like after this code executes?
+- How are `props` used within this code?
+- How does this code relate to the code within the `Student` component?
```

Comment:
I like this list of questions. Maybe add "what is being `return`ed, and where is it going?"